ui/lib/kv/addon/components/page/secret/details.js

```diff
@@ -90,24 +90,28 @@ export default class KvSecretDetails extends Component {
   async handleDestruction(type) {
     const { secret } = this.args;
     try {
-      await secret.destroyRecord({ adapterOptions: { deleteType: type, deleteVersions: secret.version } });
+      await secret.destroyRecord({ adapterOptions: { deleteType: type, deleteVersions: this.version } });
       this.store.clearDataset('kv/metadata'); // Clear out the store cache so that the metadata/list view is updated.
-      this.flashMessages.success(`Successfully ${secret.state} Version ${secret.version} of ${secret.path}.`);
+      this.flashMessages.success(`Successfully ${secret.state} Version ${this.version} of ${secret.path}.`);
       this.router.transitionTo('vault.cluster.secrets.backend.kv.secret', {
-        queryParams: { version: secret.version },
+        queryParams: { version: this.version },
       });
     } catch (err) {
       const verb = type.includes('delete') ? 'deleting' : 'destroying';
       this.flashMessages.danger(
-        `There was a problem ${verb} Version ${secret.version} of ${secret.path}. Error: ${err.errors.join(
+        `There was a problem ${verb} Version ${this.version} of ${secret.path}. Error: ${err.errors.join(
           ' '
         )}.`
       );
     }
   }
 
   get version() {
-    return this.args.secret.version || this.router.currentRoute.queryParams.version;
+    return (
+      this.args.secret?.version ||
+      this.router.currentRoute.queryParams?.version ||
+      this.args.metadata?.sortedVersions[0].version
```

Comment:
So this will always be current version? It might be clearer to pull that from the metadata model

Reply:
I could add a comment to the code clarifying but basically it's: 
- if you have read access to the secret, pull the version from there
- if you don't have read access to the secret but there's a query param for the current details view, default to that
- if there's no query param and no read access to secret, we assume it's the most current version and pull that from metadata (assuming metadata read access)
- if both metadata and secret cannot be read, the version is null because we can't read it

Reply:
Ah but we don't want to always arbitrarily pass the version, right? Oish 🥴